### PR TITLE
DRILL-7994: Dependency version updates for severe vulnerabilities

### PR DESCRIPTION
**Pull request by jnturton**
# [DRILL-7994](https://issues.apache.org/jira/browse/DRILL-7994): Dependency version updates for severe vulnerabilities

## Description

Based on the OWASP dependency check report and dependabot.

## Documentation
N/A

## Testing
Existing tests.

### CODE REVIEW

File: contrib/storage-phoenix/pom.xml [hunk 1]
```diff
@@ -33,7 +33,7 @@
     <phoenix.version>5.1.2</phoenix.version>
     <!-- Keep the 2.4.2 to reduce dependency conflict -->
     <hbase.minicluster.version>2.4.2</hbase.minicluster.version>
-    <jetty.test.version>9.4.31.v20200723</jetty.test.version>
+    <jetty.test.version>9.4.44.v20210927</jetty.test.version>
```
Comment (vdiravka) [inline contrib/storage-phoenix/pom.xml:36]: I've checked and `SecuredPhoenixTestSuite` works fine with this jetty version (dependencies in `drill-storage-phoenix` are still needed for that).

So please remove `jetty.test.version` property and it's usage

File: .travis.yml [hunk 2]
```diff
@@ -50,41 +50,43 @@ before_install:
...
   - |
-    if [ $PHASE = "tests" ]; then \
-      mvn install --batch-mode --no-transfer-progress \
-        -DexcludedGroups="org.apache.drill.categories.SlowTest,org.apache.drill.categories.UnlikelyTest,org.apache.drill.categories.SecurityTest"; \
-    elif [ $PHASE = "build_checkstyle_protobuf" ]; then \
-      MAVEN_OPTS="-Xms1G -Xmx1G" mvn install --no-transfer-progress -Drat.skip=false -Dlicense.skip=false --batch-mode -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn -DskipTests=true -Dmaven.javadoc.skip=true -Dmaven.source.skip=true && \
-      pushd protocol && mvn process-sources -P proto-compile && popd && \
-      mkdir contrib/native/client/build && pushd contrib/native/client/build && /usr/bin/cmake -G "Unix Makefiles" .. && make cpProtobufs && popd; \
+    if [ $PHASE = "tests" ]; then
+      echo Selecting parallel GC to minimise peak mem usage.
```
Comment (vvysotskyi) [inline .travis.yml:74]: Please avoid adding such comments in the ci output, they might confuse people that trying to figure out why their tests failing.

File: .travis.yml [hunk 3]
```diff
@@ -50,41 +50,43 @@ before_install:
...
+      MAVEN_OPTS="-XX:+UseParallelGC"
```
Comment (vvysotskyi) [inline .travis.yml:75]: Can we use the same GC as for regular Drill runs, so tests will be closer to the real envs? If not, why don't use UseParNewGC?

**Review by vvysotskyi — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 3 above)
Comment (jnturton) [reply .travis.yml:75]: @vdiravka it was selected for an extremely memory constrained environment and regular Drill users with such memory constraints are well-advised to make the same selection.  Will revert for now...

Comment (vdiravka) [reply .travis.yml:75]: @jnturton Could you create a task to tune the application to use less memory and back to G1 as default for GitHub CI?

**Review by vdiravka — APPROVED** [review]

**Comment by jnturton** [comment]
@vvysotskyi sorry, I wrote an @vdiravka but it was you who commented.  The idea behind this GC is to reduce memory usage in low mem environments like our CI.  The trade off is that it has some worse pause times than the newer GCs but some longer pauses are better than a crashed CI.  My prediction is that reverting it will bring back "JVM process terminated unexpectedly" for the JDK 8 build specifically.  But it can be set in another PR, I just needed a CI pass here for deps updates reasons.  It was just a means to an end for me.

**Pull request merged**